library/core/src/fmt/mod.rs

```diff
@@ -1582,8 +1599,9 @@ impl<'a> Formatter<'a> {
     /// assert_eq!(format!("{:0>8}", Foo(2)), "Foo 2");
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
+    #[inline]
     pub fn write_fmt(&mut self, fmt: Arguments<'_>) -> Result {
-        write(self.buf, fmt)
+        if let Some(s) = fmt.as_const_str() { self.buf.write_str(s) } else { write(self.buf, fmt) }
```

Comment:
This branch is observable by implementors of `fmt::Write`, right? As it will either call write_str or write_fmt, both of which can be overriden.

Reply:
@Nilstrieb why is this a problem?  The `write` method internally calls the same `self.buf.write_str` if needed.